scheduler/test/cook/test/kubernetes/controller.clj

```diff
@@ -165,6 +166,38 @@
       (with-redefs [controller/write-status-to-datomic (fn [_ status] (reset! reason (:reason status)))
                     controller/container-status->failure-reason (fn [_ _ _ _]
                                                                   (throw (ex-info "Shouldn't get called" {})))]
-        (controller/handle-pod-completed nil {:pod pod :synthesized-state {:state :pod/failed}}
+        (controller/handle-pod-completed nil "podA" {:pod pod :synthesized-state {:state :pod/failed}}
                                          :reason :reason-task-invalid))
-      (is (= :reason-task-invalid @reason)))))
\ No newline at end of file
+      (is (= :reason-task-invalid @reason))))
+
+  (testing "throws exception"
+    (let [pod (tu/pod-helper "podA" "hostA" {})
+          reason (atom nil)]
+      (.setStatus pod (V1PodStatus.))
+      (with-redefs [cc/compute-cluster-name (constantly "compute-cluster-name")
+                    controller/write-status-to-datomic (fn [_ status] (reset! reason (:reason status)))
+                    controller/container-status->failure-reason (fn [_ _ _ _]
+                                                                  (throw (ex-info "Got Exception" {})))]
+        (controller/handle-pod-completed nil "podA" {:pod pod :synthesized-state {:state :pod/failed}})
+        (is (= :reason-task-unknown @reason)))))
+
+  (testing "pod mismatch"
+    (let [pod (tu/pod-helper "podA" "hostA" {})
+          reason (atom nil)]
+      (.setStatus pod (V1PodStatus.))
+      (with-redefs [cc/compute-cluster-name (constantly "compute-cluster-name")
+                    controller/write-status-to-datomic (fn [_ status] (reset! reason (:reason status)))]
+        (try
+        (controller/handle-pod-completed nil "podB" {:pod pod :synthesized-state {:state :pod/failed}})
+        ; We expect an AssertionError
+        (is false) ; This line shouldn't execute.
+        (catch AssertionError e)))))
+  (testing "pod is nil"
```

Comment:
For consistency, newline before the new `testing`

Reply:
Ok.